Mathlib/Analysis/LocallyConvex/Bipolar.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2025 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import Mathlib.Analysis.LocallyConvex.Polar
+import Mathlib.Analysis.LocallyConvex.AbsConvex
+import Mathlib.Analysis.LocallyConvex.WeakDual
+import Mathlib.Analysis.Normed.Module.Convex
+import Mathlib.Analysis.NormedSpace.HahnBanach.Separation
+
+/-!
+
+# Bipolar Theorem
+
+## Main statements
+
+- `LinearMap.flip_polar_polar_eq`: The Bipolar Theorem: The bipolar of a set coincides with its
+  closed absolutely convex hull.
+
+## References
+
+* [Conway, *A course in functional analysis*][conway1990]
+
+## Tags
+
+bipolar, locally convex space
+-/
+
+variable {𝕜 E F : Type*}
+
+namespace LinearMap
+
+section NormedField
+
+variable [NormedField 𝕜] [NormedSpace ℝ 𝕜] [AddCommMonoid E] [AddCommMonoid F]
+variable [Module 𝕜 E] [Module 𝕜 F]
+
+variable {B : E →ₗ[𝕜] F →ₗ[𝕜] 𝕜} (s : Set E)
+
+variable [Module ℝ F] [IsScalarTower ℝ 𝕜 F] [IsScalarTower ℝ 𝕜 𝕜]
+
+theorem polar_AbsConvex : AbsConvex 𝕜 (B.polar s) := by
```

Comment:
I also wonder whether this theorem should be in either `AbsConvex.lean` or `Polar.lean`. Or would that result in large imports?

Reply:
Since being absolutely convex is a property of the polar, it seems more natural to import `AbsConvex` into `Polar` rather than the other way around, so I'm trying that.